ui/core/components/gear_picker/gear_picker.tsx

```diff
@@ -258,12 +258,13 @@ export class ItemRenderer extends Component {
 
 		newItem.allSocketColors().forEach((socketColor, gemIdx) => {
 			const gemContainer = createGemContainer(socketColor, newItem.gems[gemIdx], gemIdx);
-			if (gemIdx === newItem.numPossibleSockets - 1 && [ItemType.ItemTypeWrist, ItemType.ItemTypeHands].includes(newItem.item.type)) {
-				const updateProfession = () => {
-					gemContainer.classList[this.player.isBlacksmithing() ? 'remove' : 'add']('hide');
+			if (gemIdx === newItem.numPossibleSockets - 1 && newItem.couldHaveExtraSocket()) {
+				const updateGemSlots = () => {
+					const isBlacksmithinItemSlot = [ItemType.ItemTypeWrist, ItemType.ItemTypeHands].includes(newItem.item.type);
+					gemContainer.classList[this.player.isBlacksmithing() && isBlacksmithinItemSlot ? 'remove' : 'add']('hide');
 				};
-				this.player.professionChangeEmitter.on(updateProfession);
-				updateProfession();
+				this.player.professionChangeEmitter.on(updateGemSlots);
+				updateGemSlots();
```

Comment:
I think the old code here should still work without modification? The old code should also be higher performance, not that it matters much.